controller/aks-cluster-config-handler.go

```diff
@@ -665,6 +665,10 @@ func BuildUpstreamClusterState(ctx context.Context, secretsCache wranglerv1.Secr
 	// set addon monitoring profile
 	if addonProfile["omsAgent"] != nil {
 		upstreamSpec.Monitoring = addonProfile["omsAgent"].Enabled
+
+		if addonProfile["omsAgent"].Config == nil || len(addonProfile["omsAgent"].Config) == 0 {
```

Comment:
if `.Config` is `nil`, then it will have length of 0.
```suggestion
		if len(addonProfile["omsAgent"].Config) == 0 {
```

Reply:
Ah - Go read operations work even when data structs are nil. This is different from Java, thanks for pointing that out